packages/environment-ember-template-imports/-private/environment/preprocess.ts

```diff
@@ -1,26 +1,86 @@
 import { GlintExtensionPreprocess } from '@glint/core/config-types';
-import { parseTemplates } from 'ember-template-imports';
 import { GLOBAL_TAG, PreprocessData, TemplateLocation } from './common';
-
+/**
+ * We don't need to actually generate the code that would be emitted for
+ * a real app to build, we can use placeholders, so we have less offsets
+ * to worry about (precompile, setComponentTemplate, their imports, etc)
+ */
 const TEMPLATE_START = `[${GLOBAL_TAG}\``;
 const TEMPLATE_END = '`]';
 
+// NOTE: This import is a lie -- We are not real ESM, and this is compiled to CJS.
+//       ESM usage would be different here, and this is the CJS version of content-tag.
```

Comment:
Where's the lie here? Wouldn't this import operate the same under ESM?

In the node repl, for instance, I see this:
```
> (await import('content-tag')).Preprocessor
[class Preprocessor]
```

Reply:
imprecise language use on my part, but the content-tag code is cjs
- https://www.npmjs.com/package/content-tag?activeTab=code

and more precisely, _node_ cjs.

Which is an important distinction for this effort: https://github.com/embroider-build/content-tag/pull/18
which has more to do with balancing the tradeoff between standalone ESM package with no external node dependencies than the concept of ESM vs CJS in general.

Reply:
The import as it's currently written would work regardless of whether this module were transpiled to CJS or ESM, right? And if `content-tag` were to change that, I'd expect it to be as part of a major version bump, so I'm not sure as a reader what this comment is meant to be informing me of. What value is it adding?

Reply:
idk, my assumption has been that content-tag only supported cjs, so adding an esm (and more specifically, non-node) entrypoint would be non-breaking.

will discuss with others